### PR TITLE
Vue js version of the community website.

### PR DESCRIPTION
**Pull request by RemcoHalman**
I created a Vue version of the website. 

- Created all components
- Changed Cypress to work on `localhost:8080` instead of `127.0.0.1:8080`
- Removed the svelte files

### CODE REVIEW

**Comment by eddiejaoude** [comment]
Oh wow @RemcoHalman , we just converted it to Svelte 😱 

I do want to use VueJS (and NuxtJS) for our other projects like the `EddieBotMap` - so this will be useful for that

**Comment by RemcoHalman** [comment]
> Oh wow @RemcoHalman , we just converted it to Svelte 😱 
> 
> 
> 
> I do want to use VueJS (and NuxtJS) for our other projects like the `EddieBotMap` - so this will be useful for that

Just wanted to show the difference. And its a good practice

**Comment by eddiejaoude** [comment]
> Just wanted to show the difference. And its a good practice

I just felt bad for your time. But this is Awesome! Many of us were curious to see the different, so thank you 👍

**Comment by RemcoHalman** [comment]
@eddiejaoude Could be that someone else wants to create a React version or Angular. Could be used as a reference point 😬 just a idea tho

**Comment by AllanRegush** [comment]
If we are going to make this website with different tech. I say we use scully w/ angular & gatsby w/ react.

**Comment by eddiejaoude** [comment]
The website will probably stay in Svelte to keep it simple. But I like @RemcoHalman idea so we can see different versions. I could easily do an Angular version

**Comment by BOLT04** [comment]
Imo, I agree with continuing with Svelte and having a branch for Vue or React or Angular. But the idea is to just have different versions to see the difference, in which I agree with @RemcoHalman it's a cool idea 👍.
However, I don't think we should keep updating these branches whenever something new is developed in the `develop` branch with Svelte, or this would become hard to maintain 😅 

@RemcoHalman I'll need some time to review the Vue version and test it. Are you able to fix the merge conflicts on your side?
imagem

**Comment by RemcoHalman** [comment]
@BOLT04 I don't think that there are conflicts. 

- `src/data.js` is replaced with 2 json files `src/assets/data/<json.file>` 
- The `src/App.svelte` is now `src/main.js`

Should be no problem in the Vue version. Also only if there is a massive rebuild i would update the branch.

**Comment by BOLT04** [comment]
@RemcoHalman I understand but I can't merge this since it's not letting me through the UI 😅. So I'll test this locally later tomorrow and see what is the problem.

**Comment by RemcoHalman** [comment]
@BOLT04 If you find out the answer please let me know so that i can *try* to fix this the next time 😄

**Comment by BOLT04** [comment]
@RemcoHalman i tested it locally and I think you have the loop in the wrong div 😅 
imagem

Here is how it looks:
imagem

**Comment by RemcoHalman** [comment]
@BOLT04 changed the for loop to a `div` below should be correctly displayed now

File: src/components/Moderators.vue [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        </div>
+      </div>
+    </div>
+    <!-- END LOOP MODERATORS -->
+  </div>
+</template>
+
+<script>
+import ModeratorsData from "../assets/data/moderators.json";
+console.log(ModeratorsData);
```
Comment (BOLT04) [inline src/components/Moderators.vue:96]: I think we should remove these console logs, what do you think @RemcoHalman ?

File: src/components/Projects.vue [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      </span>
+    </div>
+    <!-- END CART ITEM -->
+  </div>
+</template>
+
+<script>
+import ProjectData from "../assets/data/projects.json";
+
+console.log(ProjectData);
```
Comment (BOLT04) [inline src/components/Projects.vue:40]:
```suggestion
```

File: src/components/Projects.vue [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+export default {
+  name: "Projects",
+  data() {
+    return {
+      ProjectData: ProjectData
```
Comment (BOLT04) [inline src/components/Projects.vue:46]: what do you think of changing to this:
```suggestion
      ProjectData
```

File: src/components/Projects.vue [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+          <h3 class="mt-3 ml-3 font-bold text-indigo-100 text-xl mb-2">
+            {{ project.name }}
+          </h3>
+          <p class="mx-3 text-indigo-100 text-base pb-3">
+            {{ project.description }}
+          </p>
+        </a>
+      </span>
+    </div>
+    <!-- END CART ITEM -->
```
Comment (BOLT04) [inline src/components/Projects.vue:33]: maybe we can make this a `Card.vue` component like in [Svelte's verion](https://github.com/EddieJaoudeCommunity/EddieJaoudeCommunity.github.io/blob/develop/src/Card.svelte). What do you think?

File: src/components/Projects.vue [hunk 5]
```diff
@@ -0,0 +1,52 @@
+<template>
+  <div id="projects">
+    <h2 class="text-center text-3xl font-semibold text-indigo-100">
+      Our Projects
+    </h2>
+
+    <div
+      v-for="projects in ProjectData"
```
Comment (BOLT04) [inline src/components/Projects.vue:8]: I think we should make variable names consistent, here it's different than `projects`. Can you change to camel case?

File: src/components/Moderators.vue [hunk 6]
```diff
@@ -0,0 +1,118 @@
+<template>
+  <div class="container my-8 p-5 bg-white rounded-lg">
+    <h2 class="text-center text-4xl mb-5 py-5 font-semibold">
+      Our Moderators
+    </h2>
+    <!-- Start for loop MODERATORS-->
+    <div v-for="moderators in ModeratorsData" :key="moderators">
```
Comment (BOLT04) [inline src/components/Moderators.vue:7]: same comment for variable naming

File: src/components/Join.vue [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+      href="https://github.com/EddieJaoudeCommunity/support/issues"
+    >
+      Join us.
+    </a>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "Join"
```
Comment (BOLT04) [inline src/components/Join.vue:30]: where is this used? Or is it the name of the component and it must be exported?

File: package.json [hunk 8]
```diff
@@ -1,38 +1,57 @@
 {
-  "name": "eddie",
-  "version": "1.0.0",
+  "name": "vuejs_version",
+  "version": "0.1.0",
```
Comment (BOLT04) [inline package.json:3]: i think we can keep the version, but this can be merged like this as well 👍

**Review by BOLT04 — COMMENTED** [review]
Hi @RemcoHalman, great work 👍. I left some comments I think we should address before merging 😃

**Comment by RemcoHalman** [comment]
@BOLT04 i think most points are changed. 

The card component is a good idea and is a change that can be made later. Just tried and would not work how i had in mind. 

The `name: "<component>"` was in every file. Thought this was needed but actually it was not. So removed them in all components. Also changed the tags in moderators from `button` to `label` since it wont link to anything 

Tried to be consistent with the lowerCamelCase but this is a personal preference. Can change this if needed

**Comment by BOLT04** [comment]
@RemcoHalman I've resolved the conflicts by accepting the deleted files on your PR 👍

**Comment by RemcoHalman** [comment]
@BOLT04 really appreciate the tips and the merge!

**Comment by eddiejaoude** [comment]
Thank you both 👍 - this way we can keep the branch in the repo.

We should probably add a note to the readme also, if people want to have a look at the vuejs version - I appreciate it will get out of date, but still gives a good comparison

**Pull request merged**